### PR TITLE
Create unified release build for macOS and Linux packages

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Create unified release build for macOS and Linux packages
  - Also, updated the definition with latest definitions from the definitions in use.

I will work on these in future PRs:

- Add compliance Job
- Add macOS Signing step
- Merging Windows into the YAML

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/releaseBuild.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  branches:
+    include:
+    - master
+    - release*
+pr:
+  branches:
+    include:
+    - master
+    - release*
+
```
Comment (anmenaga) [inline tools/releaseBuild/azureDevOps/releaseBuild.yml]: new line

**Review by anmenaga — APPROVED** [review]
Good thing.

**Comment by TravisEz13** [comment]
cc @alerickson

File: tools/releaseBuild/azureDevOps/templates/mac.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  pool: Hosted macOS Preview
+  steps:
+  #- task: <task type name>@<version>
+  #  inputs:
+  #    <task specific inputs>
+  #  displayName: '<display name of task>'
+  - powershell: |
+      tools/releaseBuild/setReleaseTag.ps1 -ReleaseTag $(ReleaseTagVar) -Variable "ReleaseTagVar"
+    displayName: 'Calculate Release Tag'
+  - task: ShellScript@2
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/mac.yml]: Can this be converted to a PowerShell task? Non-blocking for this PR.

File: tools/releaseBuild/azureDevOps/releaseBuild.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Set AzDevOps Agent to clean the machine after the end of the build
+resources:
+- repo: self
+  clean: true
+
+jobs:
+- template: templates/linux.yml
+  parameters:
+    jobName: DEB
+    buildName: ubuntu.14.04-no-appimage
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/releaseBuild.yml]: Please change this buildName

File: tools/releaseBuild/azureDevOps/releaseBuild.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    jobName: RPM
+    buildName: centos.7
+    pool: Hosted Ubuntu 1604
+
+- template: templates/linux.yml
+  parameters:
+    jobName: fxdependent
+    buildName: fxdependent
+    pool: Hosted Ubuntu 1604
+
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/releaseBuild.yml]: Please add alpine.3

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by TravisEz13** [comment]
@adityapatwardhan 	I believe I have addressed your comments

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**